src/heaps/binomial_heap.jl

```diff
@@ -0,0 +1,368 @@
+# Binomial Heap
+mutable struct MutableBinomialHeapNode{T} 
+    data::T
+    degree::Int
+    handle::Int
+    parent::Union{MutableBinomialHeapNode,Nothing}
+    child::Union{MutableBinomialHeapNode,Nothing}
+    sibling::Union{MutableBinomialHeapNode,Nothing}
+end
+MutableBinomialHeapNode(data::T, handle::Int) where T = MutableBinomialHeapNode{T}(data, 0, handle, nothing, nothing, nothing)
+#################################################
+#
+#   heap type and constructors
+#
+#################################################
+mutable struct MutableBinomialHeap{T} <: AbstractHeap{T}
+    rootList::Vector{Union{Nothing,MutableBinomialHeapNode{T}}}
+    nodecount::Int
+    cumm_nodecount::Int
+    MutableBinomialHeap{T}() where T = new{T}([], 0, 0)
+
+    function MutableBinomialHeap{T}(xs::AbstractVector{T}) where T
+        BinomialHeap = _make_binomial_heap(xs, T)
+        new{T}(BinomialHeap.rootList, BinomialHeap.nodecount, BinomialHeap.cumm_nodecount)
+    end
+    
+end
+const BinomialHeap{T} = MutableBinomialHeap{T}
+BinomialHeap(xs::AbstractVector{T}) where T = MutableBinomialHeap{T}(xs)
+
+struct HeapBoundsError <: Exception
+    msg::AbstractString
+end
+
+function show(io::IO, _heap::MutableBinomialHeap{T}) where T
+    print(io, "MutableBinomialHeap")
+    list = _heap.rootList
+    n = length(list)
+    if n == 0
+        print(io, "{$(T)}")
+    end
+    print(io, "(")
+    for i = 1:n
+        printTree(io, list[i])
+    end
+    print(io, ")")
+end
+
+function printTree(io::IO, h::Union{MutableBinomialHeapNode{T},Nothing}) where T
+    while h !== nothing 
+        print(io, h.data)
+        print(io, ",")
+        printTree(io, h.child) 
+        h = h.sibling; 
+    end
+end
```

Comment:
While this is only a matter of style, the Julia style guide recommends that functions are all lowercase with underscores between them (`print_tree`). This is also the style followed in the rest of DataStructures. That would be nice to change.
Furthermore, many of your internal functions that are mutating do not end with `!`, even though this is the convention.

Reply:
Although I tried to follow style guidelines still, I missed a few places, thanks, doing it right now